File: decompiled/sources/com/hjq/http/listener/HttpCallback.java
package com.hjq.http.listener;

import okhttp3.Call;

public class HttpCallback<T> implements OnHttpListener<T> {
    private final OnHttpListener mListener;

    public HttpCallback(OnHttpListener onHttpListener) {
        this.mListener = onHttpListener;
    }

    public void onStart(Call call) {
        OnHttpListener onHttpListener = this.mListener;
        if (onHttpListener != null) {
            onHttpListener.onStart(call);
        }
    }

    public void onSucceed(T t, boolean z) {
        onSucceed(t);
    }

    public void onSucceed(T t) {
        OnHttpListener onHttpListener = this.mListener;
        if (onHttpListener != null) {
            onHttpListener.onSucceed(t);
        }
    }

    public void onFail(Exception exc) {
        OnHttpListener onHttpListener = this.mListener;
        if (onHttpListener != null) {
            onHttpListener.onFail(exc);
        }
    }

    public void onEnd(Call call) {
        OnHttpListener onHttpListener = this.mListener;
        if (onHttpListener != null) {
            onHttpListener.onEnd(call);
        }
    }
}

File: decompiled/sources/com/hjq/http/listener/HttpCallback.kt
package com.hjq.http.listener

import okhttp3.Call

class HttpCallback<T>(
    onHttpListener: OnHttpListener<*>?
) : OnHttpListener<T> {
    private val mListener: OnHttpListener<Any?>? = onHttpListener as? OnHttpListener<Any?>

    override fun onStart(call: Call) {
        mListener?.onStart(call)
    }

    override fun onSucceed(t: T, z: Boolean) {
        onSucceed(t)
    }

    override fun onSucceed(t: T) {
        mListener?.onSucceed(t)
    }

    override fun onFail(exc: Exception) {
        mListener?.onFail(exc)
    }

    override fun onEnd(call: Call) {
        mListener?.onEnd(call)
    }
}
